### PR TITLE
CORE-1425 Add drawer for app search results

### PR DESCRIPTION
**Pull request by ash3rz**
I updated the Global Search Field a bit so that if we wanted search fields for other resource types (analyses, data, etc.) we could do that.  I only made a drawer for apps though for now.

Tested only in storybook so far.

Drawer on desktop:
image

Valid selection:
image

Sample invalid selection:
image

Mobile view:
image

### CODE REVIEW

File: public/static/locales/en/apps.json [hunk 1]
```diff
@@ -84,33 +84,42 @@
...
     "rateApp": "Rate {{appName}}",
     "rating": "Rating",
     "ratingMutationError": "Unable to update rating. Please try again.",
     "references": "References",
     "removeFromFavorites": "Remove From Favorites",
     "save": "Save",
+    "selectApps": "Select apps",
     "selectedCategoryAriaLabel": "Selected app category",
     "selectedCategoryAriaMenuItemLabel": "Selected app category menu items",
+    "selectedItems": "{{count}} selected",
```
Comment (ash3rz) [inline public/static/locales/en/apps.json:122]: My IDE sorted this file - only `selectedItems` and `selectApps` should have been added

File: src/components/search/detailed/AppSearchResults.js [hunk 2]
```diff
@@ -195,16 +206,11 @@ export default function AppSearchResults(props) {
         return <Typography>{t("noResults")}</Typography>;
     }
 
-    let flatData = [];
-    if (data && data.length > 0) {
-        data.forEach((page) => {
-            flatData = [...flatData, ...page.apps];
-        });
-    }
```
Comment (ash3rz) [inline src/components/search/detailed/AppSearchResults.js:203]: This method to get flatData on top of my changes was causing the drawer to infinitely render, so I moved this over into the `useEffect`

File: src/components/search/detailed/AppSearchDrawer.js [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+    const theme = useTheme();
+    const isMobile = useMediaQuery(theme.breakpoints.down("xs"));
+
+    const toolbarId = build(baseId, ids.SELECTION_TOOLBAR);
+
+    const errorMsg = validateSelection
+        ? validateSelection(selectedApps)
+        : false;
+
+    const Buttons = () => (
```
Comment (psarando) [inline src/components/search/detailed/AppSearchDrawer.js:86]: 👍

File: src/components/search/detailed/AppSearchDrawer.js [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+    );
+
+    return (
+        <>
+            <Toolbar id={toolbarId}>
+                {!hasSelection ? (
+                    <Typography variant="h6">{t("selectApps")}</Typography>
+                ) : errorMsg ? (
+                    <Typography
+                        style={{ color: theme.palette.error.main }}
```
Comment (psarando) [inline src/components/search/detailed/AppSearchDrawer.js:103]:
```suggestion
                        color="error"
```

File: src/components/search/detailed/AppSearchDrawer.js [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+    const [searchText, setSearchText] = useState(currentSearch);
+    const handleInputChange = (event) => setSearchText(event.target.value);
+
+    useEffect(() => {
+        let timeOutId = null;
+        if (searchText && searchText.length > 2) {
+            timeOutId = setTimeout(() => setCurrentSearch(searchText), 500);
+        }
+        return () => clearTimeout(timeOutId);
+    }, [searchText, setCurrentSearch]);
```
Comment (psarando) [inline src/components/search/detailed/AppSearchDrawer.js:138]: 👍 🎉

File: src/components/search/detailed/AppSearchDrawer.js [hunk 6]
```diff
@@ -0,0 +1,222 @@
...
+     * validateSelection is an optional function prop to check if the
+     * current selected apps meet a condition.  If the selection is invalid,
+     * it should return a string with an error message, otherwise return falsy
+     */
+    validateSelection: PropTypes.func,
+    open: PropTypes.bool.isRequired, // whether the drawer is open or not
+    onConfirm: PropTypes.func.isRequired, // when apps are selected
+    onClose: PropTypes.func.isRequired, // when the drawer is closed
+    searchTerm: PropTypes.string.isRequired, // the app search term
+};
```
Comment (psarando) [inline src/components/search/detailed/AppSearchDrawer.js:222]: 👍

**Review by psarando — APPROVED** [review]
LGTM 👍 
Some tidy refactors here as well ✨

File: src/components/search/GlobalSearchField.js [hunk 7]
```diff
@@ -145,15 +144,30 @@ const getViewAllPrompt = (id, searchTerm, i18NSearch) => {
...
                 href={href}
-                onClick={onClick}
+                onClick={(event) => {
+                    if (onOptionSelected) {
+                        event.preventDefault();
+                        onOptionSelected();
+                        return false;
+                    }
+                    // else, do default link behavior
+                }}
```
Comment (sri2k1us) [inline src/components/search/GlobalSearchField.js:170]: I think you cannot override `onClick` when doing `nextjs` client side routing. You can check this by searching for an app in global search and selecting an option. The whole page will refresh before taking you to the launch page.

**Review by sri2k1us — CHANGES_REQUESTED** [review]
Code looks good to me but I may have found a bug in nextjs client side routing. Also UX feedback for using `AppBar`.

File: src/components/search/GlobalSearchField.js (hunk 7 above)
Comment (ash3rz) [reply src/components/search/GlobalSearchField.js:175]: Good catch - I think I've got a fix in place. And the screenshots have been updated.

**Review by sri2k1us — APPROVED** [review]
:shipit:

**Comment by ash3rz** [comment]
Thanks for the reviews!

**Pull request merged**